src/sql/src/pure.rs

```diff
@@ -1523,39 +1546,113 @@ async fn purify_create_table_from_source(
             purified_export
         }
         GenericSourceConnection::LoadGenerator(load_gen_connection) => {
-            let mut views = load_gen_connection.load_generator.views();
-            // Resolve the desired view reference to a `schema_name.view_name` reference
-            let resolver = SourceReferenceResolver::new(
-                LOAD_GENERATOR_DATABASE_NAME,
-                &views
-                    .iter()
-                    .map(|(view_name, _, _)| {
-                        (load_gen_connection.load_generator.schema_name(), *view_name)
-                    })
-                    .collect_vec(),
-            )?;
-            let (qualified_reference, view_idx) = resolver.resolve(&external_reference.0, 2)?;
+            // TODO(roshan): Refactor KeyValue load generator rendering to support multiple outputs
+            if matches!(
+                load_gen_connection.load_generator,
+                mz_storage_types::sources::load_generator::LoadGenerator::KeyValue(_)
+            ) {
+                sql_bail!(
+                    "{} is a {} source, which does not support CREATE TABLE .. FROM SOURCE.",
```

Comment:
What is preventing this type of source from using `CREATE TABLE .. FROM SOURCE`?

Reply:
I didn't realize that key-value load-generators use a completely separate rendering mechanism which I haven't yet refactored to support multiple source-exports:

https://github.com/MaterializeInc/materialize/blob/f922274be328042ea3a28a6b61c352a7fb930930/src/storage/src/source/generator.rs#L144-L162

So I need to do that refactoring first before they can support added tables